oozie-to-airflow/mappers/spark_mapper.py

```diff
@@ -45,190 +59,171 @@ def __init__(
-
-        '--conf key=value'
-        '--conf key1=value1 value2'
-        '--conf key1="value1 value2"'
-        '--conf key1=value1 key2="value2 value3"'
-        '--conf key=value --verbose --properties-file user.properties'
+        --conf key1=value
+        --conf key2="value1 value2"
         """
+        conf = {}
         if spark_opts_node.text:
             spark_opts = spark_opts_node.text.split("--")[1:]
         else:
-            raise Exception("Spark opts node has no text: {}".format(spark_opts_node))
+            raise ParseException("Spark opts node has no text: {}".format(spark_opts_node))
         clean_opts = [opt.strip() for opt in spark_opts]
         clean_opts_split = [opt.split(maxsplit=1) for opt in clean_opts]
 
-        if ["verbose"] in clean_opts_split:
-            self.__dict__["verbose"] = True
-            clean_opts_split.remove(["verbose"])
-
         for spark_opt in clean_opts_split:
             # Can have multiple "--conf" in spark_opts
             if spark_opt[0] == "conf":
                 # Splits key1=value1 into [key1, value1]
-                conf_val = spark_opt[1].split("=", maxsplit=1)
-                self.conf[conf_val[0]] = conf_val[1]
-            else:
-                self.__dict__[spark_opt[0]] = "'" + " ".join(spark_opt[1:]) + "'"
+                conf_pairs = spark_opt[1].split("=", 1)
+                # Value is required
+                if len(conf_pairs) != 2:
+                    continue
+                key = conf_pairs[0]
+                value = conf_pairs[1]
+                # Delete surrounding quotes
+                if len(value) > 2 and value[0] in ["'", '"'] and value:
+                    value = value[1:-1]
+                conf[key] = value
+
+        return conf
+
+    def _get_tasks(self):
+        """
+        Returns the list of Airflow tasks that are the result of mapping
 
-    @staticmethod
-    def parse_prepare_node(prepare_node: ET.Element):
+        :return: list of Airflow tasks
         """
-        <prepare>
-            <delete path="[PATH]"/>
-            ...
-            <mkdir path="[PATH]"/>
-            ...
-        </prepare>
+        action_task = Task(
+            task_id=self.name,
+            template_name="spark.tpl",
+            template_params=dict(
+                main_jar=self.java_jar,
+                main_class=self.java_class,
+                arguments=self.application_args,
+                archives=self.hdfs_archives,
+                files=self.hdfs_files,
+                job_name=self.job_name,
+                dataproc_spark_properties=self.properties,
+                dataproc_spark_jars=self.dataproc_jars,
+                trigger_rule=self.trigger_rule,
+            ),
+        )
+
+        if not self.has_prepare(self.oozie_node):
+            return [action_task]
+
+        prepare_task = Task(
+            task_id=self.name + "_prepare",
+            template_name="prepare.tpl",
```

Comment:
In other mappers (e.g. Pig, Mapreduce) we include the prepare task in the main task's template.
If we want to do this differently maybe let's change it everywhere in a separate commit? It can even be in this PR I think, but what's important is not to have 2 ways of doing "prepare".

Reply:
I tried to avoid changes in other mappers when it is not necessary. Other mappers do not use the Task object and need to be updated to be compatible with new ideas. The change you suggest is ready. Look at: https://github.com/GoogleCloudPlatform/cloud-composer/pull/127

Reply:
Ok, thanks for the explanation.